extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -703,8 +705,17 @@ export class AmpAutocomplete extends AMP.BaseElement {
       return Promise.resolve();
     }
     const filteredData = this.filterData_(sourceData, input);
+    const dataWithConverter = filteredData.map((item) => {
+      let itemWithConverter = item;
+      // Add a function that converts the object itself to a JSON string.
+      // This function is then available in mustache templates.
+      if (typeof item === 'object') {
+        itemWithConverter = {...item, objToJson: () => JSON.stringify(item)};
+      }
```

Comment:
I'm not very familiar with mustache templates, but why should we use a function here instead of resolving it first?

Reply:
Good question. My initial thinking was that this way JSON.stringify is only run if users of the component actually add data-json={{objToJson}} to the template. This in theory improves performance when we're dealing with large objects. Otherwise, we'll run JSON.stringify for every filtered object even if the template itself doesn't need it.

This may however be premature optimization and in that case we can just inline the already built string here.

Let me know what you think. Happy to do it either way.

Reply:
Fine by me, in your experience, is working with large objects something that is likely?

Additionally, are there cases when there would be nested objects?

Reply:
Sadly I wouldn't know the answer to those questions :/
It probably depends on the use case, but I would imagine that more often than not objects will not be too large given that the data is probably already massaged for the autocomplete elements. But I have no statistics to back this up.

Would there be someone who would know? In any case, I'm happy to keep the code simpler if that's generally preferred.

Reply:
@caroqliu is there any data on how common it is to have large and/or nested objects here?

If objects are usually small and not nested, we can simplify this by not using the lazy processing approach and simply add the already parsed string.

Reply:
I'm leaving this as is for now. Feel free to reopen the discussion if the lazy approach is not needed here.